substrate/frame/message-queue/src/lib.rs

```diff
@@ -460,6 +477,10 @@ pub mod pallet {
 
 		/// Processor for a message.
 		///
+		/// Storage changes are not rolled back on error.
```

Comment:
Gav mentioned that maybe @kianenigma or @shawntabrizi could help to answer this, but we wondered whether or not this message processor should roll-back storage changes on error.  

The MQ pallet should be generic enough to do any kind of work. On one hand that would mean to have an explicit `TransactionalProcessor` wrapper type to not restrict the user, but on the other hand throughout FRAME we often assume this error rollback mechanism. WDYT?

Reply:
I do not know when that specific line was written, but it seems right that everything in the runtime assumes storage rollback when it errors.

I believe it is impossible to create a system where you have both kinds of logic, since you need to make some base level assumption about what will happen when writing code. Especially where that code can co-mingle with code where there may be different assumptions.

Reply:
Yea... I will make a follow-up issue to change that.